packages/artifact/src/internal/requestUtils.ts

```diff
@@ -0,0 +1,78 @@
+import {IHttpClientResponse} from '@actions/http-client/interfaces'
+import {
+  isRetryableStatusCode,
+  isSuccessStatusCode,
+  sleep,
+  getExponentialRetryTimeInMilliseconds
+} from './utils'
+import * as core from '@actions/core'
+import {getRetryLimit} from './config-variables'
+
+export async function retry<T>(
+  name: string,
+  operation: () => Promise<T>,
+  getStatusCode: (response: T) => number | undefined,
+  errorMessages: Map<number, string>,
+  maxAttempts: number
+): Promise<T> {
+  let response: T | undefined = undefined
+  let statusCode: number | undefined = undefined
+  let isRetryable = false
+  let errorMessage = ''
+  let extraErrorInformation: string | undefined = undefined
+  let attempt = 1
+
+  while (attempt <= maxAttempts) {
+    try {
+      response = await operation()
```

Comment:
You have removed all the `displayHttpDiagnostics` calls.  Sometimes they have valuable debug information.  How about you add it here so every http call can be traced by debug log?

Reply:
Added back the `displayHttpDiagnostics` if retries are exhausted or we hit something that can't be retried. Took a suprising amount of reworking but it's back